src/snakemake/scheduling/job_scheduler.py

```
@@ -167,6 +171,20 @@ def __init__(
             pass
         self._open_jobs.release()
 
+    def get_executor(self, job: AbstractJob) -> AbstractExecutor:
+        executor_name = job.resources["executor"]
+        assert isinstance(executor_name, str)
```

Ruff (0.12.2)

Use of `assert` detected

(S101)